predicators/approaches/bridge_policy_approach.py

```diff
@@ -486,13 +551,28 @@ def learn_from_interaction_results(
             return None
         all_states = []
         all_actions = []
-
-        for result in results:
-            new_traj = LowLevelTrajectory(result.states, result.actions)
+        policy_logs = self._policy_logs
+        for i in range(len(results)):
+            result = results[i]
+            policy_log = policy_logs[:len(result.states[:-1])]
+            mapleq_states = [
+                state for j, state in enumerate(result.states[:-1])
+                if policy_log[j] == "bridge"
+                or policy_log[max(j - 1, 0)] == "bridge"
```

Comment:
hold on - the `or` part of this statement is a bit concerning. If `j` starts at `0` and `j-1` is `-1`, wouldn't we index the list with `-1` as the first part of the `if` here, which is not what we want? Could you explain why you need this logic more?

Reply:
This is because of the way I set `self._current_control`. It ends up setting the action CallPlanner as if it was executed by the planner not the bridge, so I did this to account for the CallPlanner actions (if the action before this was done by bridge, then we set this one to bridge since I think this only happens CallPlanner is used)

Reply:
This is probably a bit bad though, lmk if you see a better way to do this

Reply:
wait - but `max(-1, 0)` is `0`, so you won't look at the state right before it?

Reply:
if j=0, then it doesn't matter since on the first time step it's always the planner's control

Reply:
ah okay, i think i get it. Please add a comment to the code explaining this in case we need to revisit in the future